### DISCUSSION TITLE
1 petabyte storage

### DISCUSSION THREAD

**Discussion by Angelo-gh3990** [seaweedfs/seaweedfs#5397] [Q&A] [answered]
Hello,

I have 1 PiB of back-end storage and would like to service this with SeaWeedFS. What would be your recommendations?
The reason I'm asking for is that every volume is 32GB, do I then have to create 32000 volumes?

Any input would be helpful, or is there an auto-scaling option here?

Because right now I have a systemd service per volume, seems like a lot of work for 32000 :-) 

Thank you

**Comment by SmsS4** [comment]
Have you looked at large disk releases?

**Reply by Angelo-gh3990** [reply]
Thanks for your reply. What do you mean by that?

**Reply by Angelo-gh3990** [reply]
@SmsS4 just wanted to reach out if you have a link or an article I can read up on  this? Thank you

**Reply by SmsS4** [reply]
I couldn't find anything related to this but you can download large_disk releases from https://github.com/seaweedfs/seaweedfs/releases
for example `linux_amd64_large_disk`

**Reply by n9** [reply]
@Angelo-gh3990 

I have found the two options in the source code:
- [4 bytes offset](https://github.com/seaweedfs/seaweedfs/blob/6aa804b368242f853b5258f20178eae8a5eda26f/weed/storage/types/offset_4bytes.go#L15-L16) = 32 GiB (normal volume size)
- [5 bytes offset](https://github.com/seaweedfs/seaweedfs/blob/6aa804b368242f853b5258f20178eae8a5eda26f/weed/storage/types/offset_5bytes.go#L15-L16) = 8 TiB (large disk)

Try to use the large disk build.

PS: If you need more, you can try adding 6 bytes offset to get 2 PiB (extra large disk).

**Reply by Angelo-gh3990** [reply]
This is great @n9 @SmsS4 , do you think I can just upgrade to this 5 byte version with the current 4 bytes and keep my data in tact?

**Reply by Angelo-gh3990** [reply]
@n9 @SmsS4 I tested it with the 8Tb large version. When I want to start the current volumes I get : 
`volume_tier.go:43 BytesOffset mismatch in volume info file`

Is there a way to convert those volumes to the right "new" BytesOffset?

**Reply by kmlebedev** [reply]
<img width="1051" alt="image" src="https://github.com/seaweedfs/seaweedfs/assets/9497591/adc16272-3074-499f-a512-ce9e8dfbf221">
It seems that you can switch from 30g to 800g safely, but it’s better to make a backup of the data before, as in all `.vif`, replace `"BytesOffset": 4` with `"BytesOffset": 5` by hand

**Reply by chrislusf** [reply]
Read this before changing to "large_disk" version:
https://github.com/seaweedfs/seaweedfs/wiki/FAQ#how-to-configure-volumes-larger-than-30gb

**Reply by Angelo-gh3990** [reply]
thank you @chrislusf , I'm stuck at :

```
To upgrade to large disk version,

remove *.idx files
use the large-disk version, run weed fix to re-generate the *.idx files
```
I removed the idx files, but when I run 
`weed fix -volumeId "ID" /weed/mnt/"ID"`
nothing happens and no new idx file is created

**Reply by Angelo-gh3990** [reply]
@kmlebedev also changing the offset in the vif files doesn't work, the service won't start throwing an error:

```
seaweedfs-volume[1946]: github.com/seaweedfs/seaweedfs/weed/storage.(*DiskLocation).loadExistingVolume(0xc000001200, {0x4d17f68, 0x>
seaweedfs-volume[1946]:         /github/workspace/weed/storage/disk_location.go:173 +0x38b
seaweedfs-volume[1946]: github.com/seaweedfs/seaweedfs/weed/storage.(*DiskLocation).concurrentLoadingVolumes.func2()
seaweedfs-volume[1946]:         /github/workspace/weed/storage/disk_location.go:213 +0xa5
seaweedfs-volume[1946]: created by github.com/seaweedfs/seaweedfs/weed/storage.(*DiskLocation).concurrentLoadingVolumes in goroutin>
seaweedfs-volume[1946]:         /github/workspace/weed/storage/disk_location.go:210 +0xcc
seaweedfs-volume[1946]: I0329 16:31:55.792504 volume_loading.go:91 readSuperBlock volume 87 version 3
seaweedfs-volume[1946]: I0329 16:31:55.792677 volume_loading.go:91 readSuperBlock volume 88 version 3
```

**Reply by SmsS4** [reply]
You have to run
```
weed fix -volumeId "ID" -collection "COLLECTION" /weed/mnt
```
or if you want to run on all volumes
```
weed fix /weed/mnt
```

**Reply by Angelo-gh3990** [reply, marked as answer]
Writing this down here so this might help people in the future :

This is the process to make it work:
```
weed shell
volume.list
```

Stop all master and volume services

then :
get every ID, and associated collection and correlate with volume directory (or you can just `ls -l` in every vol dir to see the name + ID)

in every directory:
```
rm -rf *.idx
sed -i 's/"BytesOffset": 4/"BytesOffset": 5/g' *.vif

weed fix -volumeId ID -collection COLLECTION-NAME /weed/mnt/volXXXXX
```

when I now go to the master webpage to check all is the same again with the difference being :
**SeaweedFS 8000GB** 3.64 b74e8082bac408138be99e128b8c28fd19eca7a6

///////////////////////////////////////////////////

The only thing I'm still worried about is : **Volume Size Limit	30000MB**

Any thoughts @SmsS4 @n9 @chrislusf @kmlebedev 

Very much appreciated so far everyone, really helpful and getting there step by step :-)

**Reply by chrislusf** [reply]
"The only thing I'm still worried about is : Volume Size Limit 30000MB" Now you have switched to large_disk version, you can change the master startup option to what you want.

**Comment by Angelo-gh3990** [comment]
@chrislusf I tried adding :
`weed master -volumeSizeLimitMB=8000000` earlier and it threw me an error, but I changed it back and seems to work. I must have made a typo.

Thank you!

The webinterface now shows : 
```
Volume Size Limit | 8000000MB
```

**Comment by Angelo-gh3990** [comment]
Thank you all, really appreciate it  @SmsS4 @n9 @chrislusf @kmlebedev
